Add lines and spacing to transactions page

JustinBeBoy opened the pull request:
Fix #265

Before:
<img width="801" alt="99390874-a1a6a780-28d9-11eb-9036-86fa54b8ab3b" src="https://user-images.githubusercontent.com/19331824/99636500-157db700-2a76-11eb-980b-e16efbb7de14.png">

After:
<img width="798" alt="xxxxxxxxxxxxx" src="https://user-images.githubusercontent.com/19331824/99636666-4fe75400-2a76-11eb-8182-0280199d880b.png">

ademuanthony commented:
The 'receive page' has been updated to align the QR code and the address horizontally and to hide the 'generate new address' button in a dropdown

![receive-page-amed p2 - plain](https://user-images.githubusercontent.com/11990092/55842724-b1baaa80-5b2c-11e9-868a-91cd93229623.PNG)

![receive-page-amed p2 -dropped](https://user-images.githubusercontent.com/11990092/55842740-be3f0300-5b2c-11e9-9812-6c3de8b9b65f.PNG)

ademuanthony commented:
![receive-page-qr-centered](https://user-images.githubusercontent.com/11990092/55844758-b7b58900-5b36-11e9-91d5-a18c44129bc7.PNG)

itswisdomagain commented on web/static/app/src/css/style.scss:
```diff
@@ -293,3 +293,20 @@ label.list-group-item div {
+}
+
+.result-card {
+  font-size: 15px;
```
I see a lot of elements are individually set to use font size as 15px, perhaps this can be done in body instead? Or some other parent element...
the account text and the lead text are other instances I've noticed

itswisdomagain commented on web/static/app/src/css/style.scss:
```diff
@@ -293,3 +293,20 @@ label.list-group-item div {
   display: inline !important;
 }
+
+.singleAccountTextView {
```
no longer necessary

itswisdomagain commented on web/views/receive.html:
```diff
@@ -13,54 +13,59 @@
-                               <button data-target="receive.generateNewAddressButton" data-action="click->receive#generateNewAddress"
-                                       class="btn btn-primary">Generate New Address</button>
+                               <div class="result-card">
+                                   <div data-action="click->receive#copyAddressToClipboard" style="margin-top: -20px;"
```
The margin here and in the next 2 lines, are they really necessary? I took them out and the look wasn't adversely affected imo.
If they are necessary though, comments should be added before each of those lines explaining that the negative margins are applied to reduce/eliminate some extra white space around the qr code image.

itswisdomagain commented on web/views/receive.html:
```diff
@@ -28,19 +28,20 @@ <h5 class="card-title">Receiving Decred</h5>
                            <div class="col-md-6">
-                               <div class="result-card">
+                               <div class="text-center">
+                                   {{/* the nagative margin this is to take care of the padding embedded in the qr image*/}}
```
html comments should be like this
`<!-- comment here -->`

itswisdomagain commented:
Just noticed a display issue when there are multiple accounts in wallet. The account drop down appears cut off. Something `z-index` can prolly fix.

![Screen Shot 2019-04-13 at 9 21 31 PM](https://user-images.githubusercontent.com/18400051/56084891-41fd3600-5e32-11e9-8d08-a58c630b7bd4.png)

Also, @raedah once said these dropdowns are too wide and can/should be reduced to simply fit the content adequately.

itswisdomagain commented:
Another issue with multiple accounts, this time when you select a different account to generate new address. Ideally, as soon as a different account is selected, the currently displayed address and qr code should be cleared so a user does not mistake it for the newly generated address.

In the following screenshot, the "whew" account doesn't really exist (has a wrong account number) and there was an error trying to generate the address. The error could not be displayed.

![Screen Shot 2019-04-13 at 9 24 28 PM](https://user-images.githubusercontent.com/18400051/56084949-1a5a9d80-5e33-11e9-92b2-b2b7ed1827be.png)

ademuanthony commented:
updated

![receive-page-14-4-error-page](https://user-images.githubusercontent.com/11990092/56092821-17a28b80-5eb9-11e9-8205-0f298203281f.PNG)

![receive-page-14-4-overlap-fixed](https://user-images.githubusercontent.com/11990092/56092823-1ec99980-5eb9-11e9-890b-cc6dc5bc5397.PNG)

itswisdomagain commented on web/routes/handlers.go:
```diff
@@ -170,6 +170,7 @@ func (routes *Routes) receivePage(res http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	accounts = append(accounts, accounts...)
```
I believe this is test code, please delete.

itswisdomagain commented on web/static/app/src/controllers/receive_controller.js:
```diff
@@ -1,43 +1,34 @@
   static get targets () {
     return [
-      'account', 'address', 'image', 'copyButtonText', 'generateNewAddressButton'
+      'errorMessage', 'successMessage',
```
Do we need `successMessage` on this page? We can show success messages using toastr, but error messages need to stay visible on the screen, so we should retain the `errorMessage` element

itswisdomagain commented on web/static/app/src/controllers/receive_controller.js:
```diff
@@ -1,43 +1,34 @@
     return [
-      'account', 'address', 'image', 'copyButtonText', 'generateNewAddressButton'
+      'errorMessage', 'successMessage',
+      'account', 'addressContainer', 'address', 'image', 'copyButtonText', 'generateNewAddressButton'
```
Can split this into 3 lines to semantically differentiate the sections
- account selector widget
- generated address container with qr code, address text and copy button
- generate new address button

The functions in this file should be ordered similarly:
- `getRecieveAddress` for when the account selection changes (see related comment below)
- `copyAddressToClipboard `
- `generateNewAddress`

itswisdomagain commented on web/static/app/src/controllers/receive_controller.js:
```diff
@@ -1,43 +1,34 @@
 
   generateNewAddress () {
+    hide(this.addressContainerTarget)
     this.generateNewAddressButtonTarget.textContent = 'Generating...'
```
The button's text was all small letters, should probably stick to that.

itswisdomagain commented on web/static/app/src/controllers/receive_controller.js:
```diff
@@ -46,9 +37,8 @@ export default class extends Controller {
-        if (onComplete) {
-          onComplete()
-        }
+        _this.generateNewAddressButtonTarget.textContent = 'Generate New Address'
```
This isn't the exact original button text, it was all small letters.

itswisdomagain commented on web/static/app/src/controllers/receive_controller.js:
```diff
@@ -1,43 +1,34 @@
-    let _this = this
+    const _this = this
     clearMessages(this)
     axios.get('/generate-address/' + this.accountTarget.value)
```
When the account selection changes, we don't need to generate a new address, we should return previously generated address that is unused. The `wallet.ReceiveAddress` method ensures that no new address is generated unless the previously generated address has been used. That method should be used for this scenario.

When user clicks generate new address however, the `wallet.GenerateNewAddress` method should be used to ignore previously unused addresses and return a new one.

We can make this endpoint take a `new` query parameter that can be used the route handler to determine which of the above methods to use.

dreacot committed:
chore: added 4 highlighted tabs from mockup

dreacot committed:
chore: added 4 highlighted tabs from mockup

codemaestro64 committed:
replace tabbed nav with a fixed nav drawer

codemaestro64 committed:
replace tabbed nav with a fixed nav drawer

codemaestro64 committed:
redesign transactions card

codemaestro64 committed:
redesign transactions card

dreacot committed:
chore: updated sidebar

dreacot committed:
chore: updated sidebar

dreacot committed:
chore: added icons to send a receive buttons

dreacot committed:
chore: added icons to send a receive buttons

dreacot committed:
chore: increased clickable areas for send and receive buttons

dreacot committed:
chore: increased clickable areas for send and receive buttons

dreacot committed:
feat: added more page

dreacot committed:
fix: changed icons to active depending on selected tab item

dreacot committed:
chore: rearranged icons structure on more page

dreacot committed:
chore: renamed class names

dreacot committed:
chore: made list items clickable

dreacot committed:
chore: added inner pages for more page

codemaestro64 committed:
Merge branch 'new_gui' of https://github.com/dreacot/godcr into new_gui

codemaestro64 committed:
capitalize page names

codemaestro64 committed:
add combo widget

dreacot committed:
chore: aligned topbar and side bar

codemaestro64 committed:
add filter combos to transactions list page

dreacot committed:
chore: fixed conflict

Sirmorrison committed:
fix page nav text warp

Sirmorrison committed:
fix crash when some page btn is clicked

Sirmorrison committed:
add line seperator to top nav
remove overview page hardcode unit parameters
improve combo widget to confirm to app design

Sirmorrison committed:
fix issue with popup displacing tx table downward

Sirmorrison committed:
add status icon and date to txtable ui
change collapsible widget chevron icon
remove hardcoded unit params from txpage

Sirmorrison committed:
add status icon and date to txpage
fix app crash when tx detail back btn is clicked

Sirmorrison committed:
fix lint

Sirmorrison committed:
add info icon to txdetails page
align header contents to new ui

Sirmorrison committed:
update txdetails page to newui
add step amount to tx table amount
optimize overview, tx, txdetails,page codes

Sirmorrison committed:
add info modal to txdetails page
remove modal line and title
use newui gray color on details text

Sirmorrison committed:
add blue text copy to clipboard

Sirmorrison committed:
remove commented code
rename combo to dropdown
fix lint
optimize code

JustinBeBoy committed:
add lines and spacing to transactions page
